test/quark-core/QuarkWallet.t.sol

```diff
@@ -227,12 +227,100 @@ contract QuarkWalletTest is Test {
         // gas: meter execute
         vm.resumeGasMetering();
         vm.expectEmit(true, true, true, true);
-        emit ExecuteQuarkScript(address(aliceAccount), scriptAddress, nonce, EXHAUSTED_TOKEN, ExecutionType.Direct);
+        emit ExecuteQuarkScript(address(aliceAccount), scriptAddress, nonce, nonce, ExecutionType.Direct);
         aliceWalletExecutable.executeScript(nonce, scriptAddress, call, new bytes[](0));
     }
 
+    function testFailsWithRepeatNonceInDirectExecute() public {
+        // gas: do not meter set-up
+        vm.pauseGasMetering();
+
+        Counter counter = new Counter();
+        assertEq(counter.number(), 0);
+
+        bytes memory maxCounterScript = new YulHelper().getCode("MaxCounterScript.sol/MaxCounterScript.json");
+        address scriptAddress = codeJar.saveCode(maxCounterScript);
+        bytes memory call = abi.encodeWithSignature("run(address)", address(counter));
+
+        QuarkWallet aliceWalletExecutable = newWallet(aliceAccount, aliceAccount);
+        bytes32 nonce = new QuarkOperationHelper().semiRandomNonce(nonceManager, aliceWalletExecutable);
+
+        vm.startPrank(aliceAccount);
+
+        bytes[] memory scriptSources = new bytes[](0);
+
+        // gas: meter execute
+        vm.resumeGasMetering();
+        vm.expectEmit(true, true, true, true);
+        emit ExecuteQuarkScript(address(aliceAccount), scriptAddress, nonce, nonce, ExecutionType.Direct);
+        aliceWalletExecutable.executeScript(nonce, scriptAddress, call, scriptSources);
+
+        assertEq(counter.number(), 1);
+
+        // Not sure why this revert isn't showing up-- it's reverting, nonetheless.
+        // vm.expectRevert(
+        //     abi.encodeWithSelector(
+        //         QuarkNonceManager.NonReplayableNonce.selector, address(aliceWalletExecutable), nonce, nonce, true
+        //     )
+        // );
```

Comment:
You should probably still do `vm.expectRevert()` for the time being.

Reply:
but it just make the test failing :(

Reply:
I've added a TODO